src/components/ha-icon.ts

```diff
@@ -1,37 +1,113 @@
-import { Constructor } from "../types";
+import {
+  LitElement,
+  customElement,
+  CSSResult,
+  css,
+  svg,
+  property,
+  PropertyValues,
+} from "lit-element";
 
-import "@polymer/iron-icon/iron-icon";
-// Not duplicate, this is for typing.
-// tslint:disable-next-line
-import { IronIconElement } from "@polymer/iron-icon/iron-icon";
+import { get, Store } from "idb-keyval";
 
-const ironIconClass = customElements.get("iron-icon") as Constructor<
-  IronIconElement
->;
+const iconStore = new Store("hass-icon-db", "mdi-icon-store");
+const partPromise = {};
 
-let loaded = false;
+@customElement("ha-icon")
+export class HaIcon extends LitElement {
+  @property() public icon?: string;
+  @property() public path?: string;
 
-export class HaIcon extends ironIconClass {
-  private _iconsetName?: string;
+  protected updated(changedProps: PropertyValues) {
+    if (changedProps.has("icon")) {
+      this._loadIcon();
+    }
+  }
 
-  public listen(
-    node: EventTarget | null,
-    eventName: string,
-    methodName: string
-  ): void {
-    super.listen(node, eventName, methodName);
+  protected render() {
+    if (!this.path) {
+      return;
+    }
+    return svg`
+    <svg 
+      viewBox="0 0 24 24" 
+      preserveAspectRatio="xMidYMid meet"
+      focusable="false">
+      <g>
+        <path d=${this.path}></path>
+      </g>
+    </svg>`;
+  }
 
-    if (!loaded && this._iconsetName === "mdi") {
-      loaded = true;
-      import(/* webpackChunkName: "mdi-icons" */ "../resources/mdi-icons");
+  private async _loadIcon() {
+    if (!this.icon) {
+      return;
     }
+    const iconName = this.icon.replace("mdi:", "").replace("hass:", "");
```

Comment:
What about the other iconsets ? I think that we should maybe fallback to using `iron-icon` if it's not `hass` or `mdi`?